### PR TITLE
Bring project up to current Stackable best practices

### PR DESCRIPTION
**Pull request by lfrancke**
- Antora Docs
- License change to OSL 3
- Cargo.toml: Sort keys, sort dependencies, set version to nightly
- Disable automatic CRD creation
- Move CRD YAML to /deploy/crd directory
- cargo update
- Update Github Actions
- Update Packaging
- Move to stable toolchain

### CODE REVIEW

**Review by maltesander — DISMISSED** [review]
Same thing as with kafka. Here we only have the print crd binary in the "server". We are inconsistent here. I like to have it generated and written to the file directly.
I approve for now, feel free to change it :)

File: server/Cargo.toml [hunk 1]
```diff
@@ -13,7 +13,8 @@ stackable-nifi-crd = { path = "../crd" }
 stackable-nifi-operator = { path = "../operator" }
 
 serde_yaml = "0.8"
-tokio = { version = "1", features = ["macros", "rt-multi-thread"] }
+tokio = { version = "1.6", features = ["macros", "rt-multi-thread"] }
```
Comment (soenkeliebau) [inline server/Cargo.toml:16]: Comment does not refer to this line. In other PRs you added "the main binary" to the description in this toml.

**Review by soenkeliebau — APPROVED** [review]
LGTM - left one comment

File: server/Cargo.toml (hunk 1 above)
Comment (lfrancke) [reply server/Cargo.toml:16]: True....I'm inconsistent here. Spark and ZK don't have it.
I'm too lazy to change it again everywhere now :)

**Pull request merged**